File: balanceParenthesis/stack.c
#include<stdio.h>
#include<stdlib.h>
#include "stack.h"

node* createNode(char v) {
	node* temp = malloc(sizeof(node*));
	temp->data = v;
	temp->next = NULL;
	return temp;
}

void push(stack * s, char v) {
	node* temp = *s;
	*s = createNode(v);
	(*s)->next = temp;
}

char pop(stack * s) {
	node* temp = *s;
	char v = temp->data;
	*s = temp->next;
	free(temp);
	return v;
}

int isEmpty(stack * s) {
	if(*s)
		return 0;
	return 1;
}
void init(stack * s) {
	*s = NULL;
}

File: Stack/stack.py
class Stack:
    def __init__(self):
        self.items = []
    def push(self, data):
        self.items.append(data)
    def isEmpty(self):
        return(self.items == [])
    def pop(self):
        return(self.items.pop())
    def printStack(self):
        print(self.items)
    def peek(self):
        print(self.items[-1])


s = Stack()
print(s.isEmpty())
s.push(2)
s.push(3)
s.push(4)
s.printStack()
print(s.pop())
s.printStack()
s.push('lol')
s.peek()
